lib/bap_sema/bap_sema_lift.ml

```diff
@@ -49,24 +49,22 @@ let label_of_fall cfg block =
 let annotate_insn term insn = Term.set_attr term Disasm.insn insn
 let annotate_addr term addr = Term.set_attr term address addr
 
-let linear_of_stmt ?addr return insn stmt : linear list =
+let linear_of_stmt ?addr ?dst return insn stmt : linear list =
   let (~@) t = match addr with
     | None -> t
     | Some addr -> annotate_addr (annotate_insn t insn) addr in
   let goto ?cond id =
     `Jmp ~@(Ir_jmp.create_goto ?cond (Label.direct id)) in
   let jump ?cond exp =
-    let target = Label.indirect exp in
+    let target = match dst with
```

Comment:
that doesn't look good or right. Even if it is correct, it should be rewritten in a good way.

The jump function takes exp and returns a corresponding term. Here, you're ignoring the exp if the `[dst]` parameter is passed. In this case, all jumps will lead to the same destination. 

This might be correct if `[dst]` is passed only if the statement has at most one jump. But this is a kind of precondition that you shall never enforce. It is broken.